Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -71,6 +73,7 @@ public class HTTPClient {
     private let stateLock = Lock()
 
     internal static let loggingDisabled = Logger(label: "AHC-do-not-log", factory: { _ in SwiftLogNoOpLogHandler() })
+    internal static let topLevelContextLoggingDisabled = DefaultLoggingContext(logger: loggingDisabled, baggage: .topLevel)
```

Comment:
I'm not very happy with the naming of this constant. Does anybody have any other suggestions? On a related note, I was also thinking about extending `BaggageContext` to add a factory method accepting an optional `Logger` and using this instead. That way in the `HTTPClient` method overloads taking a `Logger`, we could delegate to the matching overload taking `LoggingContext` like this:

```swift
self.get(url: url, context: .topLevel(logger: logger), deadline: deadline)
```

What do you think about this? Might be something to put in the `Baggage` repo too 🤔

Ping @ktoso

Reply:
> What do you think about this? Might be something to put in the Baggage repo too 🤔
> @slashmo

On the other hand, we also don't want to make it too easy to create empty `Baggage`s.

Reply:
> BaggageContext to add a factory method accepting an optional Logger and using this instead. 

There is no BaggageContext; You mean `LoggingContext`? Yeah I wish we could do that but we can't because a protocol, so this won't work:

```swift
 15> func take(_ p: P) {  }
 16> take(.make())
error: repl.swift:16:7: error: static member 'make' cannot be used on protocol metatype 'P.Protocol'
take(.make())
    ~~^~~~~~~
```

This API must be geared towards "pass-along whatever context you already have" which is what the LoggingContext is designed for. Passing in an "empty" or "just log" is annoying and that's for better or worse the price we pay. If someone wants just logging with no context the logger accepting overload exists today, so nothing we should try to improve about this.

--- 


> > Might be something to put in the Baggage repo too 🤔
> On the other hand, we also don't want to make it too easy to create empty Baggages.

Yeah it's on purpose that creating "empty not logging anything" is not built in. If you choose to provide that as library, that's on you.

Reply:
> If someone wants just logging with no context the logger accepting overload exists today, so nothing we should try to improve about this.
> @ktoso 

Ah, I haven't noticed the `DefaultLoggingContext.topLevel` method. With that in place I'd do:

```swift
internal static let loggingDisabledTopLevelContext = DefaultLoggingContext.topLevel(logger: loggingDisabled)
```

Or get rid of the constant and use it in place, becomes even shorter than the long constant name 😅

```swift
.topLevel(logger: loggingDisabled) // or
.topLevel(logger: logger)
```

Reply:
Ah, `.topLevel` won't work because it accepts `LoggingContext` as you mentioned. I'm using the `topLevel` method for the existing `Logger` overloads, and `loggingDisabledTopLevelContext` for the default methods.